lib/ccf/index.ts

```diff
@@ -0,0 +1,378 @@
+import {INSTANCE_TYPE_COMPUTE_PROCESSOR_MAPPING} from "@cloud-carbon-footprint/aws/dist/lib/AWSInstanceTypes";
+import {IImpactModelInterface} from "../interfaces";
+import Spline from 'typescript-cubic-spline';
+import * as aws_instances from './aws-instances.json';
+import * as gcp_instances from './gcp-instances.json';
+import * as gcp_use from './gcp-use.json';
+import * as aws_use from './aws-use.json';
+import * as azure_use from './azure-use.json';
+import * as azure_instances from './azure-instances.json';
+import * as gcp_embodied from './gcp-embodied.json';
+import * as aws_embodied from './aws-embodied.json';
+import * as azure_embodied from './azure-embodied.json';
+
+// consumption information for a single instance
+interface IConsumption {
+    idle?: number;
+    tenPercent?: number;
+    fiftyPercent?: number;
+    hundredPercent?: number;
+    minWatts?: number;
+    maxWatts?: number;
+}
+
+// information about a single compute instance
+interface IComputeInstance {
+    consumption: IConsumption;
+    embodiedEmission?: number;
+    name: string;
+    vCPUs?: number;
+    maxVCPUs?: number;
+}
+
+export enum Interpolation {
+    LINEAR = 'linear',
+    SPLINE = 'spline'
+}
+
+export class CloudCarbonFootprint implements IImpactModelInterface {
+    // Defined for compatibility. Not used in CCF.
+    authParams: object | undefined;
+    // name of the data source
+    name: string | undefined;
+    // compute instances grouped by the provider with usage data
+    computeInstances: { [key: string]: { [key: string]: IComputeInstance } } = {};
+
+    // list of all the compute instances by Architecture
+    gcpList: { [key: string]: any } = {};
+    azureList: { [key: string]: any } = {};
+    awsList: { [key: string]: any } = {};
+
+    provider: string = '';
+    instanceType: string = '';
+    expectedLifespan = 4;
+
+    interpolation = Interpolation.LINEAR;
+
+    constructor() {
+        this.standardizeInstanceMetrics();
+    }
+
+    // Defined for compatibility. Not used in CCF.
```

Comment:
We can use following format for all function documentations (JSDoc style documentation which can be used also for generating docs). It's possible to add references to docs, document params, description and so on.
```
/**
 * Defined for compatibility. Not used in CCF.
 */
authenticate(authParams: object): void {
        this.authParams = authParams;
}
```